src/config_store.rs

```diff
@@ -95,7 +95,12 @@ impl ConfigStore {
             Err(error) => {
                 error!("Unable to generate JSON from config file {}: {}",
                     file_name, error.to_string());
-                    empty_config
+                let mut backup_name = file_name.to_owned();
+                backup_name.push_str("_BROKEN");
+                warn!("Moving broken config file to {}", backup_name);
+                let _ = fs::rename(file_name, &backup_name);
```

Comment:
nit: I think you should test the return code and instead of printing the "moving" warning, print a "Moved" or "Failed to move" message (since there are lots of reasons why the rename could fail).

r=me with that change.

And as fabrice suggests, adding tests would be good.